src/Driver/RedisDriver.php

```diff
@@ -34,6 +34,17 @@ public function dequeue($queue)
     {
         list($_, $message) = $this->redis->blPop($queue, 5) ?: null;
 
-        return $message;
+        return [
+            unserialize($message),
+            null,
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function processed($job)
+    {
+        return true;
```

Comment:
Need to look into supporting: http://redis.io/commands/rpoplpush#pattern-reliable-queue